grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy

```diff
@@ -107,34 +110,46 @@ class IndicatorDataService {
         return indicatorData;
     }
 
-    DataGraph getSentStockMovements(def location) {
+DataGraph getSentStockMovements(def location, def params){
+    Integer querySize = params.querySize? params.querySize.toInteger() : 5
+    List allLocations = dataService.executeQuery("select id from openboxes.location")
+    List listDatasets = []
+    List listLabel = []
+    today.clearTime()
+    // Query data for all locations
+    for(item in allLocations){
         List listData = []
-        List listLabel = []
-        today.clearTime()
-        for(int i=5;i>=0;i--){
-            def monthBegin = today.clone()
-            def monthEnd = today.clone()
-            monthBegin.set(month: today.month - i, date: 1)
-            monthEnd.set(month: today.month - i + 1, date: 1)
-                
-            def temp = Requisition.executeQuery("""select count(r) from Requisition r where r.dateCreated >= :monthOne and r.dateCreated < :monthTwo and r.origin = :location""",
-                ['monthOne': monthBegin, 'monthTwo': monthEnd, 'location': location]);
-            String monthLabel = new java.text.DateFormatSymbols().months[monthBegin.month]
-
-            listLabel.push(monthLabel)
-            listData.push(temp[0])
+        listLabel = []
+        try{
+            List locationName = dataService.executeQuery("select name from location where id="+item[0].value)
```

Comment:
1. Why sql query, cant you use gorm? 2. Spaces around '+' operator.

Reply:
Actually, come to think of it. I might have advised you guys to use the executeQuery method. So disregard my previous comment. Just add spaces for now.